.github/workflows/ci.yaml

```diff
@@ -81,9 +82,60 @@ jobs:
       - run: yarn build:internal-dependencies
       - run: ./packages/devops/scripts/ci/validateTypesAndDbSchemaInSync.sh
 
-  run_tests:
-    name: Run tests
+  validate_lockfile:
+    name: Validate lockfile
     runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 14.19.3
+          cache: yarn
+      - run: yarn set version 3.2.1
+      - run: SKIP_BUILD_INTERNAL_DEPENDENCIES=true yarn install --immutable
+
+  package_list:
+    name: Generate package list
+    runs-on: ubuntu-latest
+    outputs:
+      PACKAGES: ${{ steps.set-matrix.outputs.PACKAGES }}
+    steps:
+      - uses: actions/checkout@v3
+      - id: set-matrix
+        run: echo PACKAGES=$(yarn workspaces list --json | jq --slurp --raw-output -c '[.[] | .name] | map(select(. != "tupaia" and . != "@tupaia/devops" and . != "@tupaia/e2e"))') >> "$GITHUB_OUTPUT"
+
+  build:
+    name: Build
+    needs:
+      package_list
+    runs-on: ubuntu-latest
+    env:
+      CI: false # react-scripts build treats warnings as errors if this is true
+
+    strategy:
+      fail-fast: false
+      matrix:
```

Comment:
The build step now builds each package in isolation leading to a speedup.

This is also needed for containerisation where each server might be its own container, we need to build that package in isolation.

This also improves our ideal of having each package declare its dependencies, for maintainability. Because only the single package and its deps are installed with `yarn workspaces focus`, it cannot rely on a module being present because a completely unrelated package installed it.